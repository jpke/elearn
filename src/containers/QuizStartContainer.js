import React from 'react';
import {connect} from 'react-redux';
// import {bindActionCreators} from 'redux';
import {startQuiz, toggleQuizView} from '../actions/eLearnActions';
import QuizStartView from '../components/QuizStartView';

export const QuizStartContainer = (props) => {

  const startQuiz = () => {
    props.startQuiz(props.token, props.quizSelectedId, props.userId);
  }

  return (
      <QuizStartView
        startQuiz={startQuiz}
        viewQuizzes={props.toggleQuizView}
        passed={props.passed}
        score={props.score}
        attempts={props.attempts}
      />
  );
};

// QuizPage.propTypes

function mapStateToProps(state) {
  return {
      score: state.quizReducer.score,
      passed: state.quizReducer.passed,
      attempts: state.quizReducer.attempts,
      quizSelectedId: state.quizReducer.quizSelectedId,
      token: state.authReducer.token,
      userId: state.authReducer._id
    };
}

function mapDispatchToProps(dispatch) {
  return {
    startQuiz: (token, quizId, userId) => dispatch(startQuiz(token, quizId, userId)),
    toggleQuizView: () => dispatch(toggleQuizView())
  };
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(QuizStartContainer);
