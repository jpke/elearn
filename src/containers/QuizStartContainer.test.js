import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureStore from '../store/configureStore';
const store = configureStore();

import QuizStartContainer from './QuizStartContainer';

it('renders correctly', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <QuizStartContainer />
    </Provider>
  ).toJSON();
  expect(tree).toMatchSnapshot();
})
