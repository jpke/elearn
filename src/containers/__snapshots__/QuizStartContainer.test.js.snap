exports[`test renders correctly 1`] = `
<div
  className="quiz-start">
  <h3>
    You have not yet attempted this quiz
  </h3>
  <div
    className="quiz-start-button-container">
    <button
      className="quiz-start-button"
      onClick={[Function]}>
       Start Quiz 
    </button>
    <button
      className="quiz-start-button"
      onClick={[Function]}>
      Back to Quiz List
    </button>
  </div>
</div>
`;
